libflux/src/core/scanner/mod.rs

```diff
@@ -174,6 +201,10 @@ impl Scanner {
             };
             match nc {
                 Some(nc) => {
+                    // It's possible that the C scanner left the data pointer in the middle
+                    // of a character. `unread_skip_whitespace()` resets the pointer to the
+                    // beginning of the character we just failed to scan.
+                    self.unread_skip_whitespace();
```

Comment:
Can you explain why whitespace matters here? Why isn't it safe to just unread the last character independent of its class?

Reply:
Good question! Originally, I just inserted a call to `unread()` and thought that was the end of it, but then [this test](https://github.com/influxdata/flux/blob/0819086f5ae0a1b13f7c3f6692de84d5b4913f16/libflux/src/core/scanner/tests.rs#L1582) was failing. `unread()` resets the scanner to the end of the last token, but if there's a whitespace character between the end of the last token and the start of the new token, a call to `unread()` will leave the pointer in the wrong location, and we will get an error when we try to [add the size of token](https://github.com/influxdata/flux/blob/0819086f5ae0a1b13f7c3f6692de84d5b4913f16/libflux/src/core/scanner/mod.rs#L179) to the pointer.

Reply:
My not directly set `self.p` to `token_start` here instead of calling `unread()`?

Reply:
That's... a good point. Hang on. Simpler fix on the way.